packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx

```diff
@@ -265,89 +266,103 @@ export class ChromeService {
     }
 
     const getHeaderComponent = () => {
-      if (chromeStyle$.getValue() === 'project') {
-        const projectNavigationComponent$ = projectNavigation.getProjectSideNavComponent$();
-        const projectBreadcrumbs$ = projectNavigation
-          .getProjectBreadcrumbs$()
-          .pipe(takeUntil(this.stop$));
-        const activeNodes$ = projectNavigation.getActiveNodes$();
-
-        const ProjectHeaderWithNavigation = () => {
-          const CustomSideNavComponent = useObservable(projectNavigationComponent$, undefined);
-          const activeNodes = useObservable(activeNodes$, []);
-
-          const currentProjectBreadcrumbs$ = projectBreadcrumbs$;
-
-          let SideNavComponent: ISideNavComponent = () => null;
-
-          if (CustomSideNavComponent !== undefined) {
-            // We have the state from the Observable
-            SideNavComponent =
-              CustomSideNavComponent.current !== null
-                ? CustomSideNavComponent.current
-                : ProjectSideNavigation;
-          }
+      const HeaderComponent = () => {
```

Comment:
I would like to move this whole component to a new file, since it takes up about 1/4 of this file. I eventually decided not to, since it is high effort due to all the state would have to be passed as props.